api/views/group.py

```diff
@@ -13,15 +13,31 @@
 from api.models import Group
 from api.processors import GroupProcessor
 from api.serializers import (
+    DashboardGroupSerializer,
+    DashboardGroupWithStudentsSerializer,
+    GroupReadSerializer,
     GroupWriteSerializer,
-    PublicGroupSerializer,
-    PublicGroupWithStudentsSerializer,
 )
+from api.views.mixins import ReadWriteSerializersMixin
 
 
-class PublicGroupViewSet(viewsets.ReadOnlyModelViewSet[Group], CreateModelMixin):
+class GroupViewSet(ReadWriteSerializersMixin, viewsets.ModelViewSet[Group]):  # type: ignore
```

Comment:
Added this `ViewSet` for API to be consistent, e.g. this view set will be used for internal needs whereas dashboard will be used only for dashboard.

Reply:
How about a `CommonGroupViewSet` with `create()` method?
And please add a short docstring :)

Reply:
Yep, but I think something like `GroupCreationMixin` would be better here, it kind of follow patter of DRF Mixins, added it to the pull request